core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -0,0 +1,388 @@
+      return namespaces;
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listNamespaces(namespace) Namespace: " + namespace, e);
+    } catch (SQLException e) {
+      throw new RuntimeException("Failed to list all namespace: " + namespace + " in catalog!", e);
+    }
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException {
+    if (!namespaceExists(namespace) || namespace.isEmpty()) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    return ImmutableMap.of("location", defaultNamespaceLocation(namespace));
+  }
+
+  private String defaultNamespaceLocation(Namespace namespace) {
+    if (namespace.isEmpty()) {
+      return warehouseLocation;
+    } else {
+      return SLASH.join(warehouseLocation, SLASH.join(namespace.levels()));
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyException {
+    if (!namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Cannot drop namespace %s because it is not found!", namespace);
+    }
+
+    List<TableIdentifier> tableIdentifiers = listTables(namespace);
+    if (tableIdentifiers != null && !tableIdentifiers.isEmpty()) {
+      throw new NamespaceNotEmptyException("Cannot drop namespace %s because it is not empty. " +
+          "Namespace contains %s tables", namespace, tableIdentifiers.size());
+    }
+
+    // namespaces are created/deleted by tables by default return true
+    // when there is no tables with namespace then its considered dropped
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) throws
+      NoSuchNamespaceException {
+    throw new UnsupportedOperationException(
+        "Cannot set properties " + namespace + " : setProperties is not supported");
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) throws NoSuchNamespaceException {
+    throw new UnsupportedOperationException(
+        "Cannot remove properties " + namespace + " : removeProperties is not supported");
+  }
+
+  @Override
+  public void close() {
+    connections.close();
+  }
+
+  @Override
+  public boolean namespaceExists(Namespace namespace) {
+    try {
+      return connections.run(conn -> {
+        boolean exists = false;
+
+        try (PreparedStatement sql = conn.prepareStatement(JdbcUtil.GET_NAMESPACE_SQL)) {
+          sql.setString(1, catalogName);
+          sql.setString(2, JdbcUtil.namespaceToString(namespace) + "%");
+          ResultSet rs = sql.executeQuery();
+          if (rs.next()) {
+            exists = true;
+          }
+
+          rs.close();
+        }
+
+        return exists;
+      });
+
+    } catch (SQLException e) {
+      LOG.warn("SQLException! ", e);
+      return false;
```

Comment:
I think this needs to throw `UncheckedSQLException`. It can't assume that any exception indicates that the table doesn't exist.

Reply:
replaced with  `throw new UncheckedSQLException(e, "Failed to get namespace %s", namespace);`